src/Avalonia.Base/Input/Platform/IClipboard.cs

```diff
@@ -6,16 +6,50 @@ namespace Avalonia.Input.Platform
     [NotClientImplementable]
     public interface IClipboard
     {
+        /// <summary>
+        /// Returns a string containing the text data on the Clipboard.
+        /// </summary>
+        /// <returns>A string containing text data in the specified data format, or an empty string if no corresponding text data is available.</returns>
         Task<string?> GetTextAsync();
 
+        /// <summary>
+        /// Stores text data on the Clipboard. The text data to store is specified as a string.
+        /// </summary>
+        /// <param name="text">A string that contains the UnicodeText data to store on the Clipboard.</param>
+        /// <exception cref="System.ArgumentNullException"><paramref name="text"/> is null.</exception>
         Task SetTextAsync(string? text);
 
+        /// <summary>
+        /// Clears any data from the system Clipboard.
+        /// </summary>
         Task ClearAsync();
 
+        /// <summary>
+        /// Places a specified non-persistent data object on the system Clipboard.
+        /// </summary>
+        /// <param name="data">A data object (an object that implements <see cref="IDataObject"/>) to place on the system Clipboard.</param>
+        /// <exception cref="System.ArgumentNullException"><paramref name="data"/> is null.</exception>
         Task SetDataObjectAsync(IDataObject data);
-        
+
+        /// <summary>
+        /// Places a specified data object on the system Clipboard and accepts a Boolean parameter that indicates whether the data object should be left on the Clipboard when the application exits.
+        /// </summary>
+        /// <param name="data">A data object (an object that implements <see cref="IDataObject"/>) to place on the system Clipboard.</param>
+        /// <param name="copy"><c>true</c> to leave the data on the system Clipboard when the application exits; false to clear the data from the system Clipboard when the application exits.</param>
+        /// <exception cref="System.ArgumentNullException"><paramref name="data"/> is null.</exception>
+        Task SetDataObjectAsync(IDataObject data, bool copy);
```

Comment:
I think correct API would be `Clipboard.FlushAsync()` which can be more reliably supported on other platforms without unnecessary throwing an exception.

Especially since we can't guarantee actual copying (or even less - not copying dataobject) on all platforms.